### PR TITLE
[ScriptingBridge] Add to Mac Catalyst. Fixes #12312.

### LINKED ISSUE #12312: [.NET 6][MacCatalyst] Add ScriptingBridge framework to MacCatalyst
### Steps to Reproduce

1. Pull [MAUI Samples](https://github.com/dotnet/maui-samples) repository;
2. Check ScriptingBridge API availability on MacCatalyst.

### Expected Behavior
ScriptingBridge API is available

### Actual Behavior
ScriptingBridge API is missing

### Environment
```
=== Visual Studio Community 2019 for Mac ===

Version 8.10.6 (build 10)
Installation UUID: 33103d27-a441-48b2-a805-c1a4a249bf20
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000140

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.140 (2020-02/51d876a041e) (64-bit)
	Package version: 612000140

=== Roslyn (Language Service) ===

3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===

Version: 5.9.0.7134

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/6.0.100-preview.6.21355.2/Sdks
SDK Versions:
	6.0.100-preview.6.21355.2
	6.0.100-preview.5.21302.13
	6.0.100-preview.4.21255.9
	6.0.100-preview.3.21202.5
	6.0.100-preview.2.21155.3
	5.0.302
	5.0.301
	5.0.203
	5.0.202
	5.0.201
	3.1.411
	3.1.410
	3.1.409
	3.1.408
	3.1.407
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	6.0.0-preview.6.21352.12
	5.0.8
	5.0.7
	5.0.6
	5.0.5
	5.0.4
	3.1.17
	3.1.16
	3.1.15
	3.1.14
	3.1.13

=== .NET Core 3.1 SDK ===

SDK: 3.1.411

=== Xamarin.Profiler ===

Version: 1.6.15.68
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Xamarin.Android ===

Version: 11.3.0.4 (Visual Studio Community)
Commit: xamarin-android/d16-10/ae14caf
Android SDK: /Users/Vasiliy_Kapitanskiy/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 30.0.4
SDK Build Tools Version: 30.0.2

Build Information: 
Mono: b4a3858
Java.Interop: xamarin/java.interop/d16-10@f39db25
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-10@c5732a0

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/Vasiliy_Kapitanskiy/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.10.0.12
Hash: e240b8c
Branch: remotes/origin/d16-10
Build date: 2021-06-01 18:26:34 UTC

=== Android Device Manager ===

Version: 16.10.0.14
Hash: e340248
Branch: remotes/origin/d16-10
Build date: 2021-06-01 18:26:52 UTC

=== Apple Developer Tools ===

Xcode 13.0 (19211.1)
Build 13A5201i

=== Xamarin.Mac ===

Version: 7.14.0.24 (Visual Studio Community)
Hash: c4b89cddb
Branch: d16-10
Build date: 2021-06-15 22:03:00-0400

=== Xamarin.iOS ===

Version: 14.20.0.24 (Visual Studio Community)
Hash: c4b89cddb
Branch: d16-10
Build date: 2021-06-15 22:03:01-0400

=== Xamarin Designer ===

Version: 16.10.0.119
Hash: 36a2d986f
Branch: remotes/origin/d16-10
Build date: 2021-06-02 19:41:34 UTC

=== Build Information ===

Release ID: 810060010
Git revision: 263d2fc9398e8092c231fa5d310cd72205cd627c
Build date: 2021-07-08 14:37:29-04
Build branch: release-8.10

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.5.0 Darwin Kernel Version 20.5.0
    Sat May  8 05:10:33 PDT 2021
    root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64
```

Android Sdks

### PR DESCRIPTION
**Pull request by rolfbjarne**
This required some xtro updates as well.

Fixes https://github.com/xamarin/xamarin-macios/issues/12312.

### CODE REVIEW

File: tools/common/Frameworks.cs [hunk 1]
```diff
@@ -650,6 +648,8 @@ public static Frameworks GetMacCatalystFrameworks ()
 
 			// Add frameworks that are not in iOS
 			catalyst_frameworks.Add ("AppKit", 13, 0);
+			catalyst_frameworks.Add ("ScriptingBridge", 13, 0);
+			catalyst_frameworks.Add ("CoreWlan", "CoreWLAN", 15, 0);
```
Comment (dalexsoto) [inline tools/common/Frameworks.cs:652]: Commit says `ScriptingBridge` any reason for `CoreWLAN` to also come over?

**Review by dalexsoto — APPROVED** [review]

File: tools/common/Frameworks.cs (hunk 1 above)
Comment (mandel-macaque) [reply tools/common/Frameworks.cs:652]: This was added in a previous commit. @rolfbjarne just moved it from line 597 to here, probably to put it with the AppKit one.  I'm guessing we had a PR race between me and @rolfbjarne

**Review by mandel-macaque — APPROVED** [review]

File: tools/common/Frameworks.cs (hunk 1 above)
Comment (rolfbjarne) [reply tools/common/Frameworks.cs:652]: > This was added in a previous commit. @rolfbjarne just moved it from line 597 to here, probably to put it with the AppKit one. I'm guessing we had a PR race between me and @rolfbjarne

Yes, that's exactly what happened.

**Comment by rolfbjarne** [comment]
It turns out that while the ScriptingBridge framework seems to be available in Mac Catalyst, none of the types in the framework are... so this PR is wrong.

**Pull request closed without merging**